### PR TITLE
Fixes uninitialised string offset

### LINKED ISSUE #394: Bug in reports.php (Add when empty)
If we click on Add without have done anything. We got an error.

Stay with the error until we click on "Remove".

capture d ecran 2017-01-14 a 16 26 20

### PR DESCRIPTION
**Pull request by asuweb**
Fixes #394

---
This change is [Reviewable](https://reviewable.io/reviews/mailwatch/1.2.0/403)

### CODE REVIEW

File: mailscanner/filter.inc.php [hunk 1]
```diff
@@ -232,7 +232,7 @@ public function CreateSQL()
                         $sql .= "AND\n $val[0] $val[1]\n";
                     } else {
                         // Allow !<sql_function>
-                        if ($val[2]{0} === '!') {
+                        if ($val[2]!='' && $val[2]{0} === '!') {
```
Comment (endelwar) [inline mailscanner/filter.inc.php]: use strict comparation `!==` and check also with `isset`

Comment (asuweb) [inline mailscanner/filter.inc.php]: Agree with the strict comparison.

Not sure I agree with checking  `$val[2]`  with isset.  In this case, `$val[2]` will always be "set" by the post data from the form, so isset will always return true in this case.

Comment (endelwar) [inline mailscanner/filter.inc.php]: yes, you're right, just the strict comparison

**Comment by stefaweb** [comment]
Can you forbid to click on Add if form "value" is empty?

File: mailscanner/filter.inc.php (hunk 1 above)
Comment (stefaweb) [inline mailscanner/filter.inc.php]: Can you forbid to click on Add if form "value" is empty?

**Comment by asuweb** [comment]
Updated with strict comparison

**Pull request merged**